tests/unit/s2n_async_pkey_test.c

```diff
@@ -142,6 +144,43 @@ int async_pkey_store_callback(struct s2n_connection *conn, struct s2n_async_pkey
     return S2N_SUCCESS;
 }
 
+struct task_params {
+    struct s2n_connection *conn;
+    struct s2n_async_pkey_op *op;
+};
+
+void *pkey_task(void *param)
+{
+    struct task_params *params = (struct task_params*)param;
+
+    struct s2n_cert_chain_and_key *chain_and_key = s2n_connection_get_selected_cert(params->conn);
+    EXPECT_NOT_NULL(chain_and_key);
+
+    s2n_cert_private_key *pkey = s2n_cert_chain_and_key_get_private_key(chain_and_key);
+    EXPECT_NOT_NULL(pkey);
+
+    EXPECT_SUCCESS(s2n_async_pkey_op_perform(params->op, pkey));
+    EXPECT_SUCCESS(s2n_async_pkey_op_apply(params->op, params->conn));
+
+    free(params);
```

Comment:
Here's your memory leak: you need to call https://github.com/aws/s2n-tls/blob/main/tls/s2n_async_pkey.c#L320 instead of directly calling free. If you're calling free in S2N, generally assume you're doing something wrong ;)

Reply:
I forgot to call `EXPECT_SUCCESS(s2n_async_pkey_op_free(params->op));`, in this case I think the call to free is OK since I am just using a struct to pass data to the background thread.

Reply:
Though I am confused why I don't see a leak in Valgrind when I run the uni test, vs what is being caught in Codebuild